myapp/app/controllers/tasks_controller.rb

```diff
@@ -2,7 +2,14 @@ class TasksController < ApplicationController
   before_action :set_task, only: [:show, :edit, :update, :destroy]
   
   def index
-    @tasks = Task.all
+    allowed_sort_columns = %w[created_at deadline]
```

Comment:
For single-use values, consider omitting the variable assignment for brevity. 
-> `%w[created_at deadline].include?(...)`, `%w[asc desc].include?(...)`

The exception is when passing a long expression as a method parameter; in that case, assigning it to a variable can improve readability. 

For `direction`, it's better to define it within the if block where it's actually used. This limits its scope, making the code cleaner and less prone to errors.